src/app/examples/crud/books.effects.spec.ts

```diff
@@ -1,77 +1,71 @@
-import { Action, Store, StoreModule } from '@ngrx/store';
-import { Observable } from 'rxjs';
-import { MockStore, provideMockStore } from '@testing/utils';
-import { State } from '../examples.state';
-import { BooksEffects, BOOKS_KEY } from './books.effects';
+import { ActionBooksDeleteOne, ActionBooksUpsertOne } from './books.actions';
 import { BookState } from './books.model';
+import { Actions, getEffectsMetadata } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { cold } from 'jasmine-marbles';
+import { EMPTY, of } from 'rxjs';
+import { BooksEffects, BOOKS_KEY } from './books.effects';
 
 describe('BooksEffects', () => {
-  const actions$: Observable<Action> = null;
-  let bookEffects: BooksEffects;
-  let metadata: EffectsMetadata<BooksEffects>;
-  let store: MockStore<State>;
-  let localStorageService: any;
-  let state: State;
-
-  const booksState: BookState = {
-    entities: {
-      '1': {
-        author: 'Author',
-        description: 'Description',
-        id: '1',
-        title: 'Title'
-      }
-    },
-    ids: ['1']
-  };
+  describe('persistBooks', () => {
+    const booksState: BookState = {
+      entities: {
+        '1': {
+          author: 'Author',
+          description: 'Description',
+          id: '1',
+          title: 'Title'
+        }
+      },
+      ids: ['1']
+    };
+    let localStorage: LocalStorageService;
+    let store: Store<any>;
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [StoreModule.forRoot({})],
-      providers: [
-        BooksEffects,
-        provideMockActions(() => actions$),
-        provideMockStore(),
-        {
-          provide: LocalStorageService,
-          useValue: jasmine.createSpyObj('LocalStorageService', ['setItem'])
+    beforeEach(() => {
+      localStorage = jasmine.createSpyObj('localStorage', ['setItem']);
+      store = of({
+        examples: {
+          books: booksState
         }
-      ]
+      }) as any;
     });
-    localStorageService = TestBed.get(LocalStorageService);
-    bookEffects = TestBed.get(BooksEffects);
-    store = TestBed.get(Store);
-    state = createState(booksState);
-    store.setState(state);
-  });
 
-  it('should be created', () => {
-    expect(bookEffects).toBeTruthy();
-  });
+    it('should not dispatch any actions', () => {
+      const actions = new Actions(EMPTY);
+      const effects = new BooksEffects(actions, store, localStorage);
+      const metadata = getEffectsMetadata(effects);
 
-  it('persistBooks should not dispatch any action', () => {
-    metadata = getEffectsMetadata(bookEffects);
-    expect(metadata.persistBooks).toEqual({ dispatch: false });
-  });
+      expect(metadata.persistBooks).toEqual({ dispatch: false });
+    });
+
+    it('should call setItem on LocalStorageService for delete one action', () => {
+      const action = new ActionBooksDeleteOne({ id: '1' });
+      const source = cold('a|', { a: action });
```

Comment:
Could we not complete the stream here, I think this could confuse some peoples in the future(?).